_posts/2023-06-11-vanilla-os-orchid-devlog.md

```diff
@@ -0,0 +1,195 @@
+---
+title: "Vanilla OS Orchid - Devlog 11 Jun"
+description: "We are happy to announce the dev log for Vanilla OS 2.0 Orchid! with various exciting updates and changes."
+published: true
+date: 2023-06-11
+layout: article
+comments: true
+---
+
+After our article [announcing](https://vanillaos.org/2023/03/07/vanilla-os-2.0-orchid-initial-work.html) the development of the second version of Vanilla OS, codenamed Orchid, several weeks of planning, development, and testing have followed. In this article, we will cover some of the new features that the upcoming major release of Vanilla OS will bring, together with our reasoning for them.
+
+## ABRoot v2 - Immutability changes, once again
+
+Throughout its lifespan, Vanilla OS has adopted [various approaches](https://vanillaos.org/2023/01/28/almost-vs-abroot.html) to achieve system immutability and atomicity from [Almost](https://github.com/Vanilla-OS/almost) to [ABRoot](https://github.com/Vanilla-OS/ABRoot). Thanks to a rapidly growing community, we have received excellent feedback. We listened to our community's opinions and learned from our mistakes. All of this has led to the conceptualization, design, and development of [ABRoot v2](https://github.com/Vanilla-OS/ABRoot), a matured version of our system that brings numerous benefits, countless new possibilities, and a higher level of stability.
+
+The new version is not just an update; the entire underlying structure has been rethought and rewritten from scratch to be the ideal solution for all use cases due to the use of OCI images.
+
+In ABRoot v1, obtaining and applying system updates was done by leveraging the package manager in an isolated and atomic environment. In ABRoot v2, the update operation now utilizes OCI images, which are exact copies of the new system, downloaded and allocated using data deduplication and applied in a single atomic transaction. This way, all installations of Vanilla OS receive the exact copy of the update, making the system 100% reproducible and faithful to our tests.
+
+It automatically raises the question, "How do I install extra drivers?" in Vanilla OS Orchid, installation of the most common drivers is a graphical and effortless process. However, there are cases where it is necessary to install additional packages in the system. For this reason, we have introduced a package manager in ABRoot v2 that allows users to manage their packages using atomic transactions.
+
+Below is an example of the commands to add and remove packages:
+
+```bash
+abroot pkg add <your-package>
+abroot pkg remove <your-package>
+```
+
+Once a package is added or removed, it will not be installed or removed from the system immediately. It will be up to the user to apply them with the command:
+
+```bash
+abroot pkg apply
+```
+
+It will create a local image of the latest Vanilla OS version, adding or removing the desired packages. This new local image would be kept up to date automatically with the latest versions of Vanilla OS.
```

Comment:
```suggestion
This will create a local image of the latest Vanilla OS version, adding or removing the desired packages. This new local image will be kept up-to-date automatically with the latest versions of Vanilla OS.
```